layer1/commons/transaction.mligo

```diff
@@ -0,0 +1,82 @@
+(* MIT License
+
+   Copyright (c) 2022 Marigold <contact@marigold.dev>
+
+   Permission is hereby granted, free of charge, to any person obtaining a copy
+   of this software and associated documentation files (the "Software"), to deal in
+   the Software without restriction, including without limitation the rights to
+   use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+   the Software, and to permit persons to whom the Software is furnished to do so,
+   subject to the following conditions:
+
+   The above copyright notice and this permission notice shall be included in all
+   copies or substantial portions of the Software.
+
+   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+   SOFTWARE. *)
+
+(** Chusai contract *)
+type chusai_contract = bytes
+
+(** Storege of Chusai contract *)
+type chusai_storage = bytes option
+
+(** Number of operations emission from Chusai contract *)
+type chusai_op_emission = nat
```

Comment:
For the first version, I was more thinking about making Chusai contract mimic Tezos contract for getting quick implement. (It can save some troubles, like get ligo work, less customize l2 interpreter etc.)

so instead of `arg * storage -> transaction list * storage`, I'd like to keep this one `arg * storage -> operation list * storage`. but I agree `transaction list` in the end goal.

If we keep this `operation list`:

- In l2, the chusai contract contains an transaction will have 2 states. The one, `s1`, is chusai contract itself. The other, `s2` is emited transaction. We can run these two one step proof separately if need.

- In l1, there is a problem for handling an `operation` type. For `s2`, the `operation list` will be empty list so no problem. But for `s1`, there is one emited transaction op in the list and  we can't pattern match `operation` to check. So the work around is that I'd check number of emited ops from `s1` and verify the result of op in `s2`.

I hope it makes sense.